src/jsontestrunner/main.cpp

```diff
@@ -126,15 +127,39 @@ static int parseAndSaveValueTree(const Json::String& input,
                                  const Json::String& actual,
                                  const Json::String& kind,
                                  const Json::Features& features, bool parseOnly,
-                                 Json::Value* root) {
-  Json::Reader reader(features);
-  bool parsingSuccessful =
-      reader.parse(input.data(), input.data() + input.size(), *root);
-  if (!parsingSuccessful) {
-    printf("Failed to parse %s file: \n%s\n", kind.c_str(),
-           reader.getFormattedErrorMessages().c_str());
-    return 1;
+                                 Json::Value* root, bool use_legacy) {
+  if (!use_legacy) {
```

Comment:
Just like @kimsey0 mentioned in #1043 

If `CharReader` has some new features that `Reader` doesn't have.There will be some questions if we do not differentiate the test/data between `CharReader` and `Reader`

Reply:
Thanks for bringing this to my attention. My suggested fix in the new revision to to prepend "legacy_" to any tests that should also run on the legacy `Reader` class. New tests that break compat should not start with "legacy_".